src/main/java/edu/ucla/library/iiif/auth/services/DatabaseServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.PoolOptions;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.SqlClient;
+import io.vertx.sqlclient.Tuple;
+
+/**
+ * The implementation of DatabaseService.
+ */
+public class DatabaseServiceImpl implements DatabaseService {
+
+    /**
+     * The database service's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseService.class, MessageCodes.BUNDLE);
+
+    /**
+     * The postgres database (and default user) name.
+     */
+    private static final String POSTGRES = "postgres";
+
+    /**
+     * The name of the database's "items" table.
+     */
+    private static final String ITEMS = "items";
+
+    /**
+     * The name of the database's "origins" table.
+     */
+    private static final String ORIGINS = "origins";
+
+    /**
+     * A single space.
+     */
+    private static final String SPACE = " ";
+
+    /**
+     * The PreparedQuery template for selecting an item's "access level".
+     */
+    private static final String SELECT_ACCESS_LEVEL = "SELECT access_level FROM items WHERE uid = $1";
+
+    /**
+     * The PreparedQuery template for upserting an item's "access level".
+     */
+    private static final String UPSERT_ACCESS_LEVEL = String.join(SPACE, "INSERT INTO items VALUES ($1, $2)",
+            "ON CONFLICT (uid) DO", "UPDATE SET access_level = EXCLUDED.access_level");
+
+    /**
+     * The PreparedQuery template for selecting an origin's "degraded allowed".
+     */
+    private static final String SELECT_DEGRADED_ALLOWED = "SELECT degraded_allowed FROM origins WHERE url = $1";
+
+    /**
+     * The PreparedQuery template for upserting an origin's "degraded allowed".
+     */
+    private static final String UPSERT_DEGRADED_ALLOWED = String.join(SPACE, "INSERT INTO origins VALUES ($1, $2)",
+            "ON CONFLICT (url) DO ", "UPDATE SET degraded_allowed = EXCLUDED.degraded_allowed");
+
+
+    /**
+     * The underlying SQL client.
+     */
+    private final SqlClient myDbClient;
+
+    /**
+     * Creates an instance of the service.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aConfig A configuration
+     */
+    public DatabaseServiceImpl(final Vertx aVertx, final JsonObject aConfig) {
+        myDbClient = PgPool.client(aVertx, getConnectionOpts(aConfig), getPoolOpts());
+    }
+
+    @Override
+    public Future<Void> close() {
+        return getSqlClient().close();
+    }
+
+    @Override
+    public Future<Integer> getAccessLevel(final String aId) {
+        final Future<RowSet<Row>> queryResult = getSqlClient().preparedQuery(SELECT_ACCESS_LEVEL)
+                .execute(Tuple.of(aId));
+
+        return queryResult.recover(error -> {
+            return Future.failedFuture(StringUtils.format(MessageCodes.AUTH_006, SELECT_ACCESS_LEVEL, error));
```

Comment:
I don't think this is doing what you think it's doing(?) The value of MessageCodes.AUTH_006 is `AUTH_006`. The usual LOGGER.getMessage(MessageCodes.CODE) method is what pulls the message associated with the `AUTH_005` key from the appropriate message bundle.

Reply:
Thanks, I'm not sure how I mixed this up

Reply:
I've done it before too.  :-)